src/coreclr/vm/riscv64/singlestepper.cpp

```diff
@@ -299,7 +299,7 @@ uint64_t RiscV64SingleStepper::GetReg(T_CONTEXT *pCtx, uint64_t reg)
 {
     _ASSERTE(reg <= 31);
 
-    return (&pCtx->R0)[reg];
+    return reg ? (&pCtx->R0)[reg] : 0; // X0 could be written/corrupted in register context, make sure it return 0.
```

Comment:
Could you explain more about this? Actually, I cannot understand why X0 could be written/corrupted and why you set hard code 0 rather than fixing corruption. cc @gbalykov

Reply:
Inside runtime we have storred registers context for frames. For example, during emulate stepping we could change some registers with JALR, but I check that register with index 0 must not change context. Looks like we also have other places too, that don't check that dest register X0 and write back some data.